### ISSUE TITLE
Need example project and tutorial

### ISSUE THREAD

**Issue by surban** [remoc-rs/remoc#5]
Although Remoc is well documented an example project and tutorial that show how all fits together and how to get started are missing.

**Comment by surban** [comment]
We now have an example project in `examples/rtc`, but more would be better.

**Comment by baptiste0928** [comment]
Implementing [tokio-rs/mini-redis](https://github.com/tokio-rs/mini-redis) with `remoc` could be a good example of real use case. I was thinking of a RTC interface with pub/sub based on broadcast channels, with automatic client reconnection and simple authentication.

This would show how all the `remoc` features fit together and give an idea of how to do your own client/server implementation.

**Comment by j-refs** [comment]
Just chiming in, as I see this issue is quite old now. The examples you mention in this repo are incomplete and non functioning.

That's definitely not welcoming to new adopters at all! 

Is there a complete/functional example somewhere that missed a merge?

**Comment by surban** [comment]
You are right that the examples are rather minimal, given the functionality of the library.

However, could you explain the trouble you have with the existing examples in the repository? They are all functioning.

**Comment by Swoorup** [comment]
This is one of the very few libraries which seems interesting requiring less ceremony at a glance, ie rpc over websockets, than worry about the internals of websocket protocol. But only a counter example doesn't do this library a justice. 😄

**Comment by surban** [comment]
We now have example projects for:
- [remote channels](https://remoc.rs/examples/channels.html)
- [remote trait calling (RPC)](https://remoc.rs/examples/rtc.html)
- [running in the browser (WebAssembly)](https://remoc.rs/examples/rtc-web.html)

Examples on how to [use transports with Remoc](https://docs.rs/remoc/latest/remoc/transports/index.html) are in the crate documentation. Furthermore [most documentation pages](https://docs.rs/remoc/latest/remoc/) have been extended with basic usage examples.

More examples are to come.
